tuna/tables_interface.py

```diff
@@ -26,32 +26,33 @@
 ###############################################################################
 """Module that encapsulates the DB representation for a library"""
 
+from typing import Dict, Set, Any, Optional
+from sqlalchemy.orm.query import Query
 from tuna.dbBase.sql_alchemy import DbSession
 
 
 #pylint: disable=too-few-public-methods
 class DBTablesInterface():
   """Represents db tables interface class"""
 
-  def __init__(self, **kwargs):
+  def __init__(self, **kwargs: Dict[str, Dict[str, Any]]):
     """Constructor"""
     super().__init__()
-    allowed_keys = set(['session_id'])
+    allowed_keys: Set = set(['session_id'])
     self.__dict__.update((key, None) for key in allowed_keys)
 
-    #for pylint
-    self.job_table = None
-    self.session_id = None
-    self.session = None
+    self.job_table: Optional[Dict[str, Any]] = None
```

Comment:
this is not a dict, this should be an option between: ConvolutionJob, BNJob and FusionJob class defined in /miopen/db/miopen_tables.py. Basically any of the classes that implement the JobMixin Interface